File: ios/Swapl/Core/Repositories/AppFeedbackRepository.swift
import Foundation

// In-app "Rate the app" feedback (F2 / M1). Posts a structured score + optional
// comment to the shared backend's AppFeedback endpoint. The same model serves
// iOS and Android; only `source` differs ("ios" here).
//
// POST /api/app-feedback expects:
//   { score: 1...5, comment?: String, source: "ios", surface: "account",
//     contextKey: "", context?: [String: …] }
final class AppFeedbackRepository: @unchecked Sendable {
    static let shared = AppFeedbackRepository()

    struct FeedbackBody: Encodable {
        let score: Int
        let comment: String?
        let source: String
        let surface: String
        let contextKey: String
    }

    @discardableResult
    func submit(
        score: Int,
        comment: String?,
        surface: String = "account",
        contextKey: String = ""
    ) async throws -> EmptyResponse {
        let trimmed = comment?.trimmingCharacters(in: .whitespacesAndNewlines)
        let body = FeedbackBody(
            score: score,
            comment: (trimmed?.isEmpty == false) ? trimmed : nil,
            source: "ios",
            surface: surface,
            contextKey: contextKey
        )
        return try await APIClient.shared.send("POST", "/api/app-feedback", body: body)
    }
}

File: android/swapl/app/src/main/java/app/swapl/core/repository/AppFeedbackRepository.kt
package app.swapl.core.repository

import app.swapl.core.network.ApiClient
import io.ktor.client.request.post
import io.ktor.client.request.setBody
import io.ktor.http.ContentType
import io.ktor.http.contentType
import kotlinx.serialization.Serializable
import javax.inject.Inject
import javax.inject.Singleton

// Structured in-app feedback (rate-the-app, M1). Shared backend model
// AppFeedback / POST /api/app-feedback serves web, iOS and Android — only the
// `source` tag differs per platform.
@Singleton
class AppFeedbackRepository @Inject constructor(private val api: ApiClient) {
    suspend fun submit(
        score: Int,
        comment: String?,
        surface: String,
        contextKey: String,
    ) {
        api.client.post("${api.baseUrl}/api/app-feedback") {
            contentType(ContentType.Application.Json)
            setBody(
                FeedbackBody(
                    score = score,
                    comment = comment,
                    source = "android",
                    surface = surface,
                    contextKey = contextKey,
                ),
            )
        }
    }

    @Serializable
    private data class FeedbackBody(
        val score: Int,
        val comment: String? = null,
        val source: String,
        val surface: String,
        val contextKey: String,
    )
}
